app/services/charges/amount_details/range_graduated_percentage_service.rb

```diff
@@ -38,11 +38,7 @@ def flat_unit_amount
       end
 
       def rate
-        @rate ||= BigDecimal(range[:rate])
-      end
-
-      def per_unit_amount
-        @per_unit_amount ||= units.zero? ? BigDecimal(0) : per_unit_total_amount.fdiv(units)
+        @rate ||= BigDecimal(range[:rate].to_s)
```

Comment:
Can `range[:rate]` be `nil`?

if yes, then we can do something like: 

```suggestion
        @rate ||= BigDecimal(range[:rate].to_s.presence || 0)
```

Reply:
At this point, `rate` should not be `nil`.

In that case, I prefer raising an error instead of putting `0` as the value